test/fuzz/random.h

```diff
@@ -0,0 +1,79 @@
+#include <random>
+#include <vector>
+
+namespace Envoy {
+namespace Random {
+
+class PsuedoRandomGenerator64 : public RandomGenerator {
+public:
+  PsuedoRandomGenerator64() = default;
+  ~PsuedoRandomGenerator64() override = default;
+
+  void initializeSeed(uint64_t seed) { prng_ = std::make_unique<std::mt19937_64>(seed); }
+
+  // RandomGenerator
+  uint64_t random() override {
+    uint64_t to_return = (*prng_.get())();
+    ENVOY_LOG_MISC(trace, "random() returned: {}", to_return);
+    return to_return;
+  }
+  std::string uuid() override { return ""; }
+  std::unique_ptr<std::mt19937_64> prng_;
+};
+
+} // namespace Random
+
+namespace Fuzz {
+class ProperSubsetSelector {
+public:
+  ProperSubsetSelector(std::string random_bytestring) {
+    random_bytestring_ = random_bytestring;
+    index_of_random_bytestring_ = 0;
+  }
+
+  // Returns a vector of vectors, each vector of bytes representing a single subset
+  std::vector<std::vector<uint8_t>>
+  constructSubsets(uint32_t number_of_subsets,
+                   std::vector<uint32_t> number_of_elements_in_each_subset,
+                   uint32_t number_of_elements) {
+    ASSERT(number_of_subsets == number_of_elements_in_each_subset.size());
+    index_vector_.clear();
+    index_vector_.reserve(number_of_elements);
+    for (uint32_t i = 0; i < number_of_elements; i++) {
+      index_vector_.push_back(i);
+    }
+    std::vector<std::vector<uint8_t>> subsets;
+    for (uint32_t i = 0; i < number_of_subsets; i++) {
+      subsets.push_back(constructSubset(number_of_elements_in_each_subset.at(i)));
+    }
+    return subsets;
+  }
+
+  // Builds a single subset by pulling indexes off index_vector_
+  std::vector<uint8_t> constructSubset(uint32_t number_of_elements_in_subset) {
```

Comment:
is this method meant to be used as a standalone? if not, make private. if so, something is weird with index_vector_.

Reply:
No, sorry. Will make private.